api/src/main/java/org/openmrs/validator/CohortValidator.java

```diff
@@ -41,13 +41,15 @@ public void validate(Object obj, Errors errors) {
 
 
 		Cohort cohort = (Cohort) obj;
-		Collection<CohortMembership> members = cohort.getMembers();
-		if (!CollectionUtils.isEmpty(members)) {
-			for (CohortMembership member : members) {
-				Patient p = member.getPatient();
-				if (p.getVoided() && !member.getVoided()) {
-					String eMessage = "Patient " + p.getPatientId() + " is voided, cannot add voided members to a cohort";
-					errors.rejectValue("members", "Cohort.patientAndMemberShouldBeVoided", eMessage);
+		if (!cohort.getVoided()) {
+			Collection<CohortMembership> members = cohort.getMemberships();
+			if (!CollectionUtils.isEmpty(members)) {
+				for (CohortMembership member : members) {
+					Patient p = member.getPatient();
+					if (p.getVoided() && !member.getVoided()) {
```

Comment:
Also reject memberships where startDate is null or if endDate is before startDate